### PR TITLE
implement client certificate in operator

### PR DESCRIPTION
**Pull request by Pespiri**

### CODE REVIEW

File: pkg/apis/deployment/ingress.go [hunk 1]
```diff
@@ -309,15 +314,57 @@ func getHostName(componentName, namespace, clustername, dnsZone string) string {
...
+			result["nginx.ingress.kubernetes.io/auth-tls-verify-client"] = string(*certificateConfig.Verification)
+			result["nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream"] = utils.TernaryString(*certificateConfig.PassCertificateToUpstream, "true", "false")
+		}
+	}
+
+	return result
+}
+
+func parseClientCertificateConfiguration(clientCertificate v1.ClientCertificate) (certificate v1.ClientCertificate) {
+	// check if public port. If yes, apply certificate conf, else remove certificate conf
```
Comment (nilsgstrabo) [inline pkg/apis/deployment/ingress.go:335]: Obsolete comment?

File: pkg/apis/utils/bools.go [hunk 2]
```diff
@@ -4,3 +4,12 @@ package utils
 func BoolPtr(value bool) *bool {
 	return &value
 }
+
+func BoolPtrCoalesce(value *bool, defaultIfNil bool) (result bool) {
```
Comment (nilsgstrabo) [inline pkg/apis/utils/bools.go:8]: This function is not used anywhere. Can be removed?

**Review by nilsgstrabo — DISMISSED** [review]

File: pkg/apis/deployment/ingress.go (hunk 1 above)
Comment (Pespiri) [reply pkg/apis/deployment/ingress.go]: I've removed this comment now

File: pkg/apis/utils/bools.go (hunk 2 above)
Comment (Pespiri) [reply pkg/apis/utils/bools.go]: Yes, this method was no longer necessary and has been removed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**